src/common/telemetry/inc/Telemetry.h

```diff
@@ -0,0 +1,127 @@
+                strcpy(filename_result, temp_template); \
+                strcat(filename_result, ".json"); \
+            } \
+        } \
+    } \
+    filename_result; \
+})
+
+// Helper macro to get module directory
+#define getModuleDirectory() ({ \
+    Dl_info dl_info; \
+    char* directory_result = NULL; \
+    if (dladdr((void*)__func__, &dl_info) != 0) { \
+        if (dl_info.dli_fname != NULL) { \
+            const char* module_fullPath = dl_info.dli_fname; \
+            const char* lastSlash = strrchr(module_fullPath, '/'); \
+            if (lastSlash != NULL) { \
+                size_t dirLen = lastSlash - module_fullPath; \
+                directory_result = (char*)malloc(dirLen + 1); \
+                if (directory_result) { \
+                    strncpy(directory_result, module_fullPath, dirLen); \
+                    directory_result[dirLen] = '\0'; \
+                } \
+            } \
+        } \
+    } \
+    directory_result; \
+})
+
+// Buffer sizes for string conversion of numeric values
+// Based on maximum possible digits for each type plus sign and null terminator
+#define MAX_INT_STRING_LENGTH 16    // Accommodates 32-bit int values
+#define MAX_LONG_STRING_LENGTH 32   // Accommodates 64-bit long values
+
+// Static file handle for telemetry logging with thread safety
+static FILE* g_telemetryFile __attribute__((unused)) = NULL;
+static char* g_fileName __attribute__((unused)) = NULL;
+static char* g_moduleDirectory __attribute__((unused)) = NULL;
+static int g_telemetryFileInitialized __attribute__((unused)) = 0;
+
+// Cleanup telemetry file - automatically called at program exit
+#define OSConfigProcessTelemetryFile() do { \
+    if (g_telemetryFile && g_fileName && g_moduleDirectory) { \
+        char* command = FormatAllocateString("%s/%s -v %s %d", g_moduleDirectory, TELEMETRY_BINARY_NAME, g_fileName, TELEMETRY_TIMEOUT_SECONDS); \
+        if (NULL != command) \
+        { \
+            ExecuteCommand(NULL, command, false, false, 0, TELEMETRY_TIMEOUT_SECONDS, NULL, NULL, NULL); \
+        } \
+        FREE_MEMORY(command); \
+    } \
+    g_telemetryFile = NULL; \
+    g_telemetryFileInitialized = 0; \
+    FREE_MEMORY(g_fileName); \
+    FREE_MEMORY(g_moduleDirectory); \
+} while(0)
+
+// Initialize telemetry file - call once at program start (thread-safe)
+#define OSConfigTelemetryInit() do { \
+    if (!g_telemetryFileInitialized) { \
+        static int initLock = 0; \
+        if (__sync_bool_compare_and_swap(&initLock, 0, 1)) { \
+            if (!g_telemetryFile) { \
+                g_moduleDirectory = getModuleDirectory(); \
+                g_fileName = generateRandomFilename(); \
+                g_telemetryFile = fopen(g_fileName, "a"); \
+                if (g_telemetryFile) { \
+                    g_telemetryFileInitialized = 1; \
+                } \
+            } \
+        } else { \
+            while (!g_telemetryFileInitialized) { \
+                usleep(1000); \
```

Comment:
Don't!!!! This will block the client. When out entire ASB audit can happen in 1 sec or faster, waiting here like this?

Please remove this.

This is blocking